qrenderdoc/Windows/Dialogs/CaptureDialog.cpp

```diff
@@ -1117,9 +1155,24 @@ void CaptureDialog::UpdateRemoteHost()
   const RemoteHost *host = m_Ctx.Replay().CurrentRemote();
 
   if(host && host->IsADB())
+  {
     ui->cmdLineLabel->setText(tr("Intent Arguments"));
+    ui->activityLabel->show();
+    ui->activity->show();
+    ui->activityBrowse->show();
```

Comment:
Having the activity browse be hidden for non-Android users is good, but even for Android users the vast majority are unlikely to need this feature so I don't think it's best to have it as a separate browser. Given that no actual developers have reported this issue it doesn't seem to be urgent so I think as much as possible it should only be intrusive when really needed.

It seems like it's already intuitive or recognisable to treat activities as somehow files within a package, since it is represented as "com.packagename.foo/Activity.name". I think it would be better to show packages as directories when browsing and then have activities as files within them.

The only problem is that then users who have well-configured apps with only one Activity still have to go down one level of the hierarchy to then select the default activity for no good reason, but this is a more minor interference and the UI will guide the users to do that anyway. That could also be improved incrementally if there's a way to smooth that experience without changing it.

Reply:
By default, the launcher Activity is displayed in the Activity input field. The other Activities are requested only when the user clicks the Activity browser button. 

If we put this function in the package browser, the processing would run in the UI thread and if the users want to launch the app with the original launcher Activity, they would indeed have to make unnecessary steps.

Reply:
Yes I understand how it works, as I've mentioned above there needs to be a solution that isn't so slow that it causes such UI problems. You're also assuming a bad implementation by saying it _must_ happen on the UI thread, which is not the case.

I agree that it's not ideal for users to have to redundantly select an Activity when almost always this doesn't matter. This is the best trade-off in my view as it's not much of a concession and with more work I think the solution can be improved.

However it's a fair point that this feature has not been requested by any real developer or user, so perhaps it should not be implemented until that happens so we can understand the usage better.